Source/Dafny/Resolver.cs

```diff
@@ -17338,6 +17401,30 @@ class DefaultValueSubstituter : Substituter {
           }
           continue;
         }
+        if (conjunct is FunctionCallExpr fce) {
```

Comment:
I don't like to ask this, but could you follow-up with moving all bounds-related code from Resolver.cs to a separate file?

Also, could you put this new block of code in a separate method?